internal/partial/partial.go

```diff
@@ -65,32 +65,44 @@ type WithLayers interface {
 	Layers() ([]v1.Layer, error)
 }
 
-func GGUFPath(i WithLayers) (string, error) {
-	return layerPathByMediaType(i, types.MediaTypeGGUF)
+func GGUFPaths(i WithLayers) ([]string, error) {
+	return layerPathsByMediaType(i, types.MediaTypeGGUF)
 }
 
 func MMPROJPath(i WithLayers) (string, error) {
-	return layerPathByMediaType(i, types.MediaTypeMultimodalProjector)
+	paths, err := layerPathsByMediaType(i, types.MediaTypeMultimodalProjector)
+	if err != nil {
+		return "", fmt.Errorf("get mmproj layer paths: %w", err)
+	}
+	if len(paths) == 0 {
+		return "", fmt.Errorf("model does not contain any layer of type %q", types.MediaTypeMultimodalProjector)
+	}
+	if len(paths) > 1 {
+		return "", fmt.Errorf("found %d files of type %q, expected exactly 1",
+			len(paths), types.MediaTypeMultimodalProjector)
+	}
+	return paths[0], err
 }
 
-// layerPathByMediaType is a generic helper function that finds a layer by media type and returns its path
-func layerPathByMediaType(i WithLayers, mediaType ggcr.MediaType) (string, error) {
+// layerPathsByMediaType is a generic helper function that finds a layer by media type and returns its path
+func layerPathsByMediaType(i WithLayers, mediaType ggcr.MediaType) ([]string, error) {
 	layers, err := i.Layers()
 	if err != nil {
-		return "", fmt.Errorf("get layers: %w", err)
+		return nil, fmt.Errorf("get layers: %w", err)
 	}
+	var paths []string
```

Comment:
```suggestion
	paths := make([]string, 0, len(layers))
```
Not sure if this is a suggestion worth accepting.  It might be slightly more optimal for large shard counts where most layers are shards.  Not sure though, you probably have a better feel for it.